enhancements/network/host-port-registry.md

```diff
@@ -23,35 +23,71 @@ This document serves as the authoritative list of port assignments.
+tools.
 
-The Kubernetes scheduler is aware of host ports, and will fail to schedule pods
-on a node where there would be a port conflict.
+Other than the reserved ranges and the other ports listed below, host
+ports on worker nodes are available for use by customers. (But all
+host ports on masters are reserved for OCP use.)
 
-### Localhost
+The Kubernetes scheduler is aware of host ports, and will fail to
+schedule pods on a node where there would be a port conflict.
 
-Separately, it is a common pattern for a pod to have one or more processes only
-listening on localhost. Since all HostNetwork pods share a network namespace,
-coordination is also required in this case.
+## Configuration
 
-## Requirements
+The installer's terraform templates open the 9000-9999 range [and
+several
+others](https://github.com/openshift/openshift-docs/blob/master/modules/installation-network-user-infra.adoc)
+between nodes in IPI clusters. Customers are required to ensure these
+ranges are open in UPI clusters. The user-facing documentation,
+informed by the installer's terraform templates, is the authoritative
+source for reserved port ranges.
+
+## Requirements for OCP Components
 
 All processes that wish to listen on a host port MUST
 
-- Have an entry in the table below
+- have an entry in the table below
 - be in a [documented range](https://github.com/openshift/openshift-docs/blob/master/modules/installation-network-user-infra.adoc),
-  if they are intended to be reachable
+  if they are intended to be reachable from other nodes
 - declare that port in their `Pod.Spec`
 
-Localhost-only ports SHOULD be outside of this range, to leave room.
+Starting with OCP 4.8, all new worker-node host port allocations
+SHOULD be in the reserved 9000-9999 port range, and all new
+worker-node localhost-only port allocation SHOULD be in the
+29000-29999 range, to avoid conflicting with ports claimed by customer
+pods created in earlier releases. If there is a strong argument for
+using a port outside these ranges by default, the relevant operator
+MUST have a configuration option to allow a customer to move the
+service into the reserved port range (or to a port of the customer's
+choice) instead.
+
+## Requirements for Customers
+
+Customers should not claim host ports in the reserved range
+(9000-9999), or any of the ports listed below. Additionally, customers
+may not claim any host ports on the OCP masters.
+
+Any customer pod that claims a host port must declare that port in its
+`Pod.Spec` so that future conflicts can be detected. Customers should
+not run non-containerized services that claim host ports on OCP nodes.
+
+In cases where a customer wants to deploy an application that uses a
+port in the 9000-9999 or 29000-29999 range by default, and the port
+that it uses is not already in use by OCP, the customer can ask us to
+mark that port as reserved for customer use. (At the present time, no
+such ports are reserved.)
```

Comment:
Do we want to put any constraints on this at all? Such as, only commonly available open source components that are likely to be adopted by multiple customers?